test/upgrade-db.sh

```
@@ -0,0 +1,192 @@
+# -q suppresses the "INSERT 0 1" command tag, which would otherwise be
+# concatenated onto the returned id.
+NEWID="$(docker compose exec -T db psql -qtA -U postgres -d rallly \
+  -c "INSERT INTO polls (title) VALUES ('post-upgrade') RETURNING id" 2>/dev/null | tr -d '[:space:]')"
+[ "$NEWID" = "4" ] && ok "sequence continues correctly (new id=$NEWID)" \
+  || bad "expected next id 4, got '$NEWID' — sequence not restored"
+
+grep -q '^POSTGRES_VERSION=18$'                        .env && ok "POSTGRES_VERSION pinned to 18"      || bad "POSTGRES_VERSION not 18"
+grep -q '^POSTGRES_DATA_MOUNT=/var/lib/postgresql$'    .env && ok "POSTGRES_DATA_MOUNT updated for 18" || bad "POSTGRES_DATA_MOUNT wrong"
+grep -q '^POSTGRES_VOLUME=db-data-pg18$'               .env && ok "POSTGRES_VOLUME switched"           || bad "POSTGRES_VOLUME wrong"
+[ "$(grep -c '^POSTGRES_VERSION=' .env)" = "1" ] && ok "no duplicate POSTGRES_VERSION lines" || bad "duplicate POSTGRES_VERSION lines"
+[ "$(stat -f '%Lp' .env 2>/dev/null || stat -c '%a' .env)" = "600" ] && ok ".env still mode 600" || bad ".env permissions changed"
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)